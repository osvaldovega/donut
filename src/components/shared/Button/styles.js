import { StyleSheet } from 'react-native';
import Colors from '../../../constants/Colors';

const styles = StyleSheet.create({
  wrapper: {
    width: '100%',
    height: 'auto',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 15,
    marginBottom: 15,
  },
  text: {
    width: 200,
    fontSize: 18,
    fontFamily: 'montserrat',
    fontWeight: '600',
    padding: 5,
    borderRadius: 9,
    overflow: 'hidden',
    backgroundColor: Colors.clearChill,
    color: Colors.antiFlashWhite,
    textAlign: 'center',
  },
});

export default styles;
