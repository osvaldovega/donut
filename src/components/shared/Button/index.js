import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import styles from './styles';

const Button = ({
  styleContainer,
  styleText,
  text,
  onPressFn,
}) => (
    <TouchableOpacity style={[styles.wrapper, styleContainer]} onPress={onPressFn}>
      <Text style={[styles.text, styleText]}>{text}</Text>
    </TouchableOpacity>
);

Button.defaultProps = {
  styleContainer: '',
  styleText: '',
  text: '',
  onPressFn: () => {},
};

export default Button;
