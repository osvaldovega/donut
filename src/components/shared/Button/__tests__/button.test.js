import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import Button from '../index';

describe('<Button />', () => {

  const setup = (props = null) =>  {
    return props
    ? renderer.create(<Button {...props} />)
    : renderer.create(<Button />)
  };

  it('renders correctly', () => {
    const instance = setup();
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a prop value', () => {
    const newProps = { text: 'Click Me' };
    const instance = setup(newProps);
    const textIns = instance.root.findByType(Text);
    expect(textIns.props.children).toBe(newProps.text);
  });
});
