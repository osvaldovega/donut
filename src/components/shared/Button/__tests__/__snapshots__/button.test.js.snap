// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "auto",
      "justifyContent": "center",
      "marginBottom": 15,
      "marginTop": 15,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "#1e90ff",
          "borderRadius": 9,
          "color": "#f1f2f6",
          "fontFamily": "montserrat",
          "fontSize": 18,
          "fontWeight": "600",
          "overflow": "hidden",
          "padding": 5,
          "textAlign": "center",
          "width": 200,
        },
        "",
      ]
    }
  >
    
  </Text>
</View>
`;
